website/blog/2023-02-21-ferretdb-compare-arrays.md

```diff
@@ -0,0 +1,380 @@
+
+So `[3, 1, 'b']` is less than `[3, 'b', 'a']`.
+
+:::note
+If an array has no more element to compare, the one with no more element is less.
+For the same reason, an empty array is less than arrays with elements.
+:::
+
+## Filter operator comparison
+
+In this section, we use `mongosh` to show examples of the comparison in filtering.
+
+### `$lt` operator comparison
+
+[$lt](https://docs.ferretdb.io/basic_operations/read/#retrieve-documents-using-operator-queries)
+filter operators select records that are less than the query.
+We use the following collection to demonstrate the `$lt` filter argument.
+
+```js
+db.ltexample.insertMany([
+  { _id: 'scalar-null', v: null },
+  { _id: 'scalar-number', v: 1 },
+  { _id: 'scalar-string', v: 'a' },
+  { _id: 'array-empty', v: [] },
+  { _id: 'array-null', v: [ null ] },
+  { _id: 'array-mixed', v: [ 1, 'z', 'b' ] },
+  { _id: 'array-string', v: [ 'd' ] },
+  { _id: 'array-bool', v: [ false ] },
+  { _id: 'nested-empty', v: [ [] ] },
+  { _id: 'nested-mixed', v: [ [ 'c' ], [ 1, 'z', 'b' ] ] },
+  { _id: 'nested-string', v: [ [ 'g' ] ] },
+  { _id: 'nested-bool', v: [ [ true ] ] }
+])
+```
+
+**Example:** `$lt` Operator with Scalar Argument
+
+When the filter argument is a string, documents with string records
+that has less value would be selected.
+
+```js
+db.ltexample.find({v: {$lt: 'c'}})
+```
+
+The output:
+
+```js
+[
+  { _id: 'scalar-string', v: 'a' },
+  { _id: 'array-mixed', v: [ 1, 'z', 'b' ] }
+]
+```
+
+The documents selected are `scalar-string` and `array-mixed`.
+
+The `scalar-string` document has a smaller value compared to the filter argument.
+
+The array record `array-mixed` was selected because the smallest element `b` is
+less than the filter argument `c`.
+
+If a document does not contain the same BSON type as the filter argument such as `array-null`, no selection is made.
+
+**Example:** `$lt` Operator with Array Argument
+
+When an array is used as the filter argument, documents with array fields containing values that are less than the values in the filter array are selected.
+It also selects nested arrays if the inner array is less than the filter argument.
+
+The following example applies the `$lt` operator with an array argument `['c']`.
+
+```js
+db.ltexample.find({v: {$lt: ['c']}})
+```
+
+The output:
+
+```js
+[
+  { _id: 'array-empty', v: [] },
+  { _id: 'array-null', v: [ null ] },
+  { _id: 'array-mixed', v: [ 1, 'z', 'b' ] },
+  { _id: 'nested-empty', v: [ [] ] },
+  { _id: 'nested-mixed', v: [ [ 'c' ], [ 1, 'z', 'b' ] ] }
+]
+```
+
+No scalar values were selected, because array argument only selects arrays.
+The filter argument selected arrays that are less
+and nested arrays which contain an array that is less.
+
+So _contains_ condition is applied on documents containing less than the filter argument, but not the other way around.
```

Comment:
This condition needs to be explained in the earlier section with example I think.

Reply:
I can't seem to find where this was mentioned

Reply:
It's this one https://github.com/FerretDB/FerretDB/blob/bc761c53bb0ba74f2ce9e10870c6a223c6382c58/website/blog/2023-03-01.ferretdb-compare-arrays.md?plain=1#L123

Reply:
Maybe we need $eq to explain it. https://github.com/FerretDB/FerretDB/blob/bc761c53bb0ba74f2ce9e10870c6a223c6382c58/website/blog/2023-03-01.ferretdb-compare-arrays.md?plain=1#L240

Reply:
I'm hoping we don't expand it too much as-is. So, can we simply explain it in a sentence or two what that contain condition does without necessarily providing an example?